src/tools/x/src/main.rs

```diff
@@ -1,9 +1,27 @@
 use std::{
-    env,
-    os::unix::process::CommandExt,
-    process::{self, Command},
+    env, io,
+    process::{self, Command, ExitStatus},
 };
 
+trait CommandExt {
+    fn exec_or_status(&mut self) -> io::Result<ExitStatus>;
+}
+
+#[cfg(unix)]
+impl CommandExt for Command {
+    fn exec_or_status(&mut self) -> io::Result<ExitStatus> {
+        use std::os::unix::process::CommandExt;
+        Err(self.exec())
+    }
+}
+
+#[cfg(not(unix))]
+impl CommandExt for Command {
+    fn exec_or_status(&mut self) -> io::Result<ExitStatus> {
+        self.status()
+    }
+}
```

Comment:
This doesn't need to be a trait I think, you can just have two free functions.

```suggestion
#[cfg(unix)]
fn exec_or_status(&mut self) -> io::Result<ExitStatus> {
    use std::os::unix::process::CommandExt;
    Err(self.exec())
}

#[cfg(not(unix))]
fn exec_or_status(&mut self) -> io::Result<ExitStatus> {
    self.status()
}
```

Reply:
fixed